### PR TITLE
Pytest profiles

### LINKED ISSUE #137: Support multiple profiles with tests
For running locally and for more CI jobs.

### PR DESCRIPTION
**Pull request by Emiller88**
Closes #137

### CODE REVIEW

File: tests/config/nextflow.config [hunk 1]
```diff
@@ -1,20 +1,15 @@
-
 params {
     outdir = "output/"
     publish_dir_mode = "copy"
     enable_conda = false
```
Comment (drpatelh) [inline tests/config/nextflow.config:4]:
```suggestion
    enable_conda = false
    singularity_pull_docker_container = false
```

**Comment by Emiller88** [comment]
Alright @drpatelh it's ready. I can't access the logs to see why singularity and conda are failing on gh-actions. Up to you whether you want those to be fixed or not before merging. If it's something simple I vote fix it so all of the other PRs have to fix their singularity and conda stuff if theirs issues and the modules repo doesn't just inherit all of the broken CI jobs.

File: tests/config/nextflow.config (hunk 1 above)
Comment (Emiller88) [reply tests/config/nextflow.config:4]: Do you want me to re-enable singularity?

**Comment by drpatelh** [comment]
This looks amazing @Emiller88 ! So the Singularity / Conda tests aren't running?

**Review by drpatelh — CHANGES_REQUESTED** [review]

File: tests/config/nextflow.config [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
-    }
+if("$PROFILE" == "docker") {
+    docker.enabled = true
+    docker.runOptions = '-u \$(id -u):\$(id -g)'
+} else if ("$PROFILE" == "singularity") {
+    singularity.enabled = true
+    singularity.autoMounts = true
+} else if ("$PROFILE" == "conda") {
+    params.enable_conda = true
 }
```
[the remark this answers is not in the document]
Comment (drpatelh) [reply tests/config/nextflow.config:16]: Make Docker the default as you suggested.

**Comment by Emiller88** [comment]
> This looks amazing @Emiller88 ! So the Singularity / Conda tests aren't running?

Thanks! No they're failing they should upload an artifact, but I don't have rights to access those.

File: tests/config/nextflow.config (hunk 2 above)
[the remark this answers is not in the document]
Comment (Emiller88) [reply tests/config/nextflow.config:16]: I was thinking about this case. Wouldn't it break if you tried to use something like `-profile crick` to test?

**Comment by drpatelh** [comment]
Ok. Let all all of the tests fire off for all 3 and I will have a look to find anything useful.

File: tests/config/nextflow.config (hunk 2 above)
[the remark this answers is not in the document]
Comment (drpatelh) [reply tests/config/nextflow.config:16]: It most likely would be we don't have the rest of the fluff to pull those configs anyway so shouldn't matter. Hopefully, there should be a `singularity` / `docker not found` error that will be obvious. 

We basically need to add some very clear documentation to the main README!

**Comment by Emiller88** [comment]
Okay I fixed the conda issue, I just had to add the channels for bioconda. 

Good/bad news. It's picking up irregularities already. https://github.com/nf-core/modules/runs/1818575675. Look at the bowtie2_align for docker and conda. The md5s don't match up for conda. I'm guessing a slight version difference.

File: tests/config/nextflow.config [hunk 3]
```diff
@@ -1,20 +1,16 @@
...
-        singularity.enabled = true
-        singularity.autoMounts = true
-    }
+if("$PROFILE" == "docker") {
+    docker.enabled = true
+    docker.runOptions = '-u \$(id -u):\$(id -g)'
+} else if ("$PROFILE" == "singularity") {
+    singularity.enabled = true
+    singularity.autoMounts = true
+} else if ("$PROFILE" == "conda") {
```
Comment (drpatelh) [inline tests/config/nextflow.config:14]:
```suggestion
    singularity.autoMounts = true
    singularity.runOptions = '-B /tmp/`
} else if ("$PROFILE" == "conda") {
```

Comment (drpatelh) [reply tests/config/nextflow.config:14]: Can you try pushing this commit @Emiller88 ?

File: tests/config/nextflow.config (hunk 2 above)
[the remark this answers is not in the document]
Comment (Emiller88) [reply tests/config/nextflow.config]: That's fine with me then. We can always go back. #139 should fix that.

**Review by drpatelh — APPROVED** [review]
Thanks a bunch @Emiller88 ! I want to try a couple of things to see if we can get the Conda / Singularity tests working.

**Pull request merged**